### PR TITLE
Update outdated docs for torch logs

### PR DESCRIPTION
**Pull request by nluu175**
Fixes #137285

## Description
Update Python code to better address the usage of python logging and `torch.compile`.

## Checklist
- [x] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [x] Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

cc @williamwen42 @msaroufim @anijain2305

### CODE REVIEW

**Comment by nluu175** [comment]
@ezyang could you please give me some feedbacks on this fix?

**Comment by ezyang** [comment]
TBH, maybe we should introduce tlparse here, it's much better for looking at the output code bits. Description at https://docs.google.com/document/d/1y5CRfMLdwEoF1nTk9q8qEu1mgMUuUtvhklPKJ2emLU8/edit?tab=t.0#heading=h.u9711c2s6w0c

**Comment by nluu175** [comment]
> TBH, maybe we should introduce tlparse here, it's much better for looking at the output code bits. Description at https://docs.google.com/document/d/1y5CRfMLdwEoF1nTk9q8qEu1mgMUuUtvhklPKJ2emLU8/edit?tab=t.0#heading=h.u9711c2s6w0c

Thank you! I'll take a look at it.

**Comment by ezyang** [comment]
waiting for render

File: recipes_source/torch_logs.py [hunk 1]
```diff
@@ -31,52 +31,81 @@
...
+######################################################################
+# Use TORCH_TRACE/tlparse to produce produce compilation reports
+# ~~~~~~~~~~
+#
+# In this section, we introduce the usage of TORCH_TRACE and tlparse to produce reports.
+# First, we run `TORCH_TRACE="/tmp/tracedir" python script.py` to generate the raw trace logs.
+# We have replace `/tmp/tracedir` with a path to a directory you want to store the trace logs
+# and reaplce script with the name of your script.
+# 
+# Next, we are going to pass the trace log to `tlparse` to generate compilation reports. We run
```
Comment (svekars) [inline recipes_source/torch_logs.py:103]:
```suggestion
# 2. Install ``tlparse`` by running:
#
#    .. code-block:: bash
#
#       pip install tlparse
#
# 3. Pass the trace log to ``tlparse`` to generate compilation reports:
#
#     .. code-block: bash
#
#        tlparse /tmp/tracedir
#
```

File: recipes_source/torch_logs.py [hunk 2]
```diff
@@ -31,52 +31,81 @@
...
+# `pip install tlparse` and then `tlparse /tmp/tracedir`. This will open up your browser with 
+# HTML like generated above.
```
Comment (svekars) [inline recipes_source/torch_logs.py:105]:
```suggestion
# This will open your browser with the HTML-like generated above.  
#
```

File: recipes_source/torch_logs.py [hunk 3]
```diff
@@ -31,52 +31,81 @@
...
+#
+# By default, reports generated by `tlparse` are
```
Comment (svekars) [inline recipes_source/torch_logs.py:107]:
```suggestion
# By default, reports generated by ``tlparse`` are 
```

File: recipes_source/torch_logs.py [hunk 4]
```diff
@@ -31,52 +31,81 @@
...
+# First, we run `TORCH_TRACE="/tmp/tracedir" python script.py` to generate the raw trace logs.
+# We have replace `/tmp/tracedir` with a path to a directory you want to store the trace logs
+# and reaplce script with the name of your script.
+# 
+# Next, we are going to pass the trace log to `tlparse` to generate compilation reports. We run
+# `pip install tlparse` and then `tlparse /tmp/tracedir`. This will open up your browser with 
+# HTML like generated above.
+#
+# By default, reports generated by `tlparse` are 
+# stored in the directory `tl_out`. You can change that by doing `tlparse /tmp/tracedir -o output_dir/`.
```
Comment (svekars) [inline recipes_source/torch_logs.py:108]:
```suggestion
# stored in the ``tl_out`` directory. You can change that by running:
#  
# .. code-block:: bash
#
#    tlparse /tmp/tracedir -o output_dir/
```

File: recipes_source/torch_logs.py [hunk 5]
```diff
@@ -31,52 +31,81 @@
...
-if torch.cuda.get_device_capability() < (7, 0):
-    print("Skipping because torch.compile is not supported on this device.")
-else:
-    @torch.compile()
-    def fn(x, y):
-        z = x + y
-        return z + 2
 
+def env_setup():
+    """Set up environment for running the example. Exit cleanly if CUDA is not available."""
```
Comment (svekars) [inline recipes_source/torch_logs.py:38]:
```suggestion
    """Set up the environment to run the example. Exit cleanly if CUDA is not available."""
```

File: recipes_source/torch_logs.py [hunk 6]
```diff
@@ -31,52 +31,81 @@
...
+        sys.exit(0)
+        
+    if torch.cuda.get_device_capability() < (7, 0):
+        print("Skipping because torch.compile is not supported on this device.")
+        sys.exit(0)
 
-    inputs = (torch.ones(2, 2, device="cuda"), torch.zeros(2, 2, device="cuda"))
 
+def separator(name):
+    """Print separator and reset dynamo between each example"""
```
Comment (svekars) [inline recipes_source/torch_logs.py:49]:
```suggestion
    """Print a separator and reset dynamo between each example"""
```

**Review by svekars — COMMENTED** [review]
A few editorial suggestions.

File: recipes_source/torch_logs.py [hunk 7]
```diff
@@ -31,52 +31,102 @@
...
+#
+#       pip install tlparse
+#
+# 3. Pass the trace log to ``tlparse`` to generate compilation reports:
+#
+#     .. code-block: bash
+#
+#        tlparse /tmp/tracedir
+#
+# This will open your browser with the HTML-like generated above.
```
Comment (svekars) [inline recipes_source/torch_logs.py:122]:
```suggestion
# This will open your browser with the HTML-like code generated above.  
```

**Comment by nluu175** [comment]
@ezyang hello. Do I need to made any more modifications for this issue? I see that the pipeline is still waiting to be reported.
